### ISSUE TITLE
[CT-2727] [Bug] dbt docs serve edits the manifest.json

### ISSUE THREAD

**Issue by Nisond** [dbt-labs/dbt-core#7922]
### Is this a new bug in dbt-core?

- [X] I believe this is a new bug in dbt-core
- [X] I have searched the existing issues, and I could not find an existing issue for this bug

### Current Behavior

When I run dbt docs serve --port 8000 it edits the manifest.json, the index.html, and the partial_parse.msgpack in the target folder.

I have verified that this edit happens by looking at the last modified timestamp on the file as well as by manually making edits to the json which then are reverted upon running docs serve.

### Expected Behavior

dbt docs serve --port 8000 to not edit any files in the target folder

### Steps To Reproduce

Unclear if other are able to replicate this. I'm using dbt-core=1.5.0 and dbt-bigquery=1.5.0 on linux.

These are steps I can take to replicate 
1. opened a terminal
2. activate the virtual environment I use for dbt, containing dbt-core and dbt-bigquery
3. used cd to navigate to the dbt project folder
4. ran ‘dbt docs serve --port 8000’

I have a single terminal open on the local machine and am not hosting documentation. I have tried restarting the machine first.

### Relevant log output

```shell
N/A
```

### Environment

```markdown
- OS: Debian 4.19.0-24-cloud-amd64
- Python: 3.7.12
- dbt:1.5.0
```

### Which database adapter are you using with dbt?

bigquery

### Additional Context

See [here](https://discourse.getdbt.com/t/dbt-docs-serve-edits-manifest-json/8727/9) for original issue and discussion.

**Comment by dbeatty10** [comment]
Thank you for opening this issue @Nisond !

I read through the [discussion](https://discourse.getdbt.com/t/dbt-docs-serve-edits-manifest-json/8727) in Discourse that you linked -- thank you for including that helpful context. 🧠

It sounds like your concrete goal in this instance is to do post-processing of `manifest.json` and add hyperlinks to model descriptions. While trying to accomplish that, it sounds like you were surprised that `dbt docs serve` doesn't just use the `manifest.json` as-is.

## Follow-up questions
Could you give an example of a model description that you'd be adding a hyperlink to?

Also, what would be the barriers to adding those hyperlinks to the descriptions directly like below?

### Before

`models/_models.yml`
```yaml
models:
  - name: legendary_creatures
    description: This description doesn't have any hyperlinks
```

### After

`models/_models.yml`
```yaml
models:
  - name: legendary_creatures
    description: >
      [`Legendary_creatures`](https://en.wikipedia.org/wiki/Category:Legendary_creatures) is a category on Wikipedia.
      [Deadpool](https://en.wikipedia.org/wiki/Deadpool) was listed in the top-10 as-of 2023-06-22 ([ref](https://en.wikipedia.org/wiki/Wikipedia:Popular_pages#Legendary_Creatures)).
```

**Comment by joellabes** [comment]
I think this might be resolved by https://github.com/dbt-labs/dbt-core/pull/7554?

**Comment by dbeatty10** [comment]
@Nisond we believe this was resolved by https://github.com/dbt-labs/dbt-core/pull/7554 and available in [dbt-core 1.5.1](https://github.com/dbt-labs/dbt-core/blob/ed92f1369cabe0d8fe1e61c897aa7ded6a241e91/CHANGELOG.md#dbt-core-151---may-30-2023) and onwards.

Could you give that a shot and let us know if you run into any problems?
